lib/core/src/test_utils/persist.rs

```diff
@@ -123,12 +123,26 @@ pub(crate) fn new_receive_swap(
 
 macro_rules! create_persister {
     ($name:ident) => {
-        let temp_dir = tempdir::TempDir::new("liquid-sdk")?;
+        #[cfg(all(target_family = "wasm", target_os = "unknown"))]
+        let temp_dir_path = {
+            use rand::Rng;
+            let res: String = rand::thread_rng()
+                .sample_iter(&rand::distributions::Alphanumeric)
+                .take(16)
+                .map(char::from)
+                .collect();
+            res
+        };
+
+        #[cfg(not(all(target_family = "wasm", target_os = "unknown")))]
+        let temp_dir_path = tempdir::TempDir::new("liquid-sdk")?
```

Comment:
@danielgranhao currently we are including/excluding wasm related code using macros. I think on cases where we have a clear trait and implementation we can utilize the approach of passing the implementation from outside instead.
We already having the LiquidSdkBuilder which builds the sdk instance and has the ability to config services such persister, onchain wallet, etc..
We can implement the connect there and even if it means some code duplication it might be a cleaner approach.
LMK what you think.